templates/install_puppet.bat.erb

```diff
@@ -0,0 +1,11 @@
+:wait_for_pid
+FOR /F "tokens=*" %%A IN ('tasklist /FI "PID eq <%= @puppet_agent_pid %>"  /NH') DO set _task=%%A
+echo %_task% | findstr "No tasks are running" >nul
+IF NOT %errorlevel% == 0 ( GOTO wait_for_pid )
```

Comment:
not sure sleep is required... if the ruby process id is not in the tasklist, then neither it nor any children should be running. Also sleeping in batch files isn't easy... a common idiom is to `ping` localhost for N times waiting 1 second in between.